### PR TITLE
Comments: prettify state + data-layer

### PR DESCRIPTION
**Pull request by samouri**
Commands ran: 
```bash
./node_modules/prettier/bin/prettier.js --write client/state/comments/*.js* 
./node_modules/prettier/bin/prettier.js --write client/state/comments/**/*.js
./node_modules/prettier/bin/prettier.js --write client/state/data-layer/wpcom/comments/*.js*
./node_modules/prettier/bin/prettier.js --write client/state/data-layer/wpcom/comments/**/*.js*
./node_modules/prettier/bin/prettier.js --write client/state/data-layer/wpcom/sites/comments/**/*.js*
```

### CODE REVIEW

**Comment by samouri** [comment]
Has two lint warning.  I'm okay with them

**Comment by dmsnell** [comment]
Do we need to investigate and report these lint errors? cc: @jsnajdr?

**Comment by gwwar** [comment]
Looks like it's just whitespace? Maybe a corner case for our fork. @samouri do you mind tidying this up manually for now?
```
/home/ubuntu/wp-calypso/client/state/data-layer/wpcom/sites/comments/index.js
  63:6  error  Expected indentation of 4 tabs but found 5  indent
  64:5  error  Expected indentation of 3 tabs but found 4  indent
```

**Comment by samouri** [comment]
> Looks like it's just whitespace? Maybe a corner case for our fork

Yep! I'd be okay with fixing it manually right now.  In terms of doing forward though: we have a few options here:
1. update our fork to diverge in another location so that it has the level of indentation that conforms to our style guide for this corner case
2. modify our style guide to conform with prettier

I vote for the latter :)

**Comment by jsnajdr** [comment]
Prettier and ESLint's `indent` rule have different opinions on how to indent expressions inside ternaries:

Prettier:
```js
const error = site
  ? translate("Failed to retrieve comments for site “%(siteName)s”", {
      args: { siteName: site.name }
    })
  : translate("Failed to retrieve comments for your site");
```
ESLint:
```js
const error = site
  ? translate("Failed to retrieve comments for site “%(siteName)s”", {
    args: { siteName: site.name }
  })
  : translate("Failed to retrieve comments for your site");
```
Some links to Prettier and ESLint's issues where this is discussed:
https://github.com/eslint/eslint/issues/7698
https://github.com/eslint/eslint/issues/6606
https://github.com/prettier/prettier/issues/1271

My opinion is that Prettier does it right and that the ESLint rule should get a configuration option.

The fact that we have a fork doesn't make any difference here -- we don't touch this part of the formatting algorithm.

**Comment by dmsnell** [comment]
Let's take all formatting rules out of eslint!

**Comment by samouri** [comment]
yes. but lets merge this before then 😆

File: client/state/comments/actions.js [hunk 1]
```diff
@@ -130,21 +130,33 @@ export function changeCommentStatus( siteId, postId, commentId, status ) {
 			siteId,
 			postId,
 			commentId,
+<<<<<<< HEAD
 			status
+=======
+>>>>>>> prettify state + data-layer
```
Comment (Copons) [inline client/state/comments/actions.js]: @samouri I believe this should be the only rebase conflict introduced by my PR.
I just added `status` there (https://github.com/Automattic/wp-calypso/pull/16082/files#diff-a1e5174c2173e0927694ec3bd26d79d4), nothing major.

**Comment by samouri** [comment]
Thanks for fixing this up @bluefuton !

File: client/state/comments/actions.js (hunk 1 above)
Comment (bluefuton) [inline client/state/comments/actions.js]: Thanks @Copons! Fixed up now 👍

**Comment by kwight** [comment]
I love this. I noticed two strange things:
* dangling `=`s: https://github.com/Automattic/wp-calypso/pull/16116/files#diff-16e50fc2140c911fab6cee0fd8dc385eR99
* arrow functions with simple returns being spread across mutliple lines: https://github.com/Automattic/wp-calypso/pull/16116/files#diff-1f1818e52092939a106eee5eaef1da11R18

But I don't see any reason why not to land this (smoke tests were fine), and for us to start using `calypso-prettier` for all comments-related code.

**Comment by samouri** [comment]
> I love this. I noticed two strange things:

- Dangling commas: I agree with this.  I'm a bigger fan of the previous styling here
- arrow functions split across multiple lines: The old line was ~140 characters.  It was violating our style guide.  Splitting here was the right move

**Pull request merged**